src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,46 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int k = 0;
+        int [] diagonal = new int [matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[k] = matrix[i][i];
```

Comment:
```suggestion
            diagonal[i] = matrix[i][i];
```
Let's removing `k` variable